include/zephyr/drivers/lora.h

```diff
@@ -87,6 +87,12 @@ struct lora_modem_config {
 	/** Set to true for transmission, false for receiving */
 	bool tx;
 
+	/** Channel activity detection enable */
+	bool cad;
+
+	/** Maximum random delay in milliseconds before channel activity detection */
+	uint32_t cad_delay_max;
```

Comment:
Why is the random delay part of the API at all? Surely if someone wants this functionality that can simply delay before calling the TX function?